runhouse/servers/http/http_server.py

```diff
@@ -204,6 +204,8 @@ def get_cert():
     def check_server():
         try:
             HTTPServer.register_activity()
+            if not ray.is_initialized():
+                raise Exception("Ray is not initialized, restart the server.")
```

Comment:
FYI: was observing failures where Ray didn't start properly which were not being caught to trigger restart_server. Just doing this is enough to make sure it's triggered (and takes practically no time at all).